proto/gossip.proto

```diff
@@ -44,6 +46,61 @@ message GossipRequest {
 message GossipResponse {
   // Requested delta of server's infostore.
   optional bytes delta = 1;
+  // TODO(thschroeter): replace bytes with GossipInfoStore
+  //optional GossipInfoStore delta = 1;
   // Non-nil means client should retry with this address.
   optional Addr alternate = 2;
 }
+
+message GossipInfoStore {
+  // Map from key to info
+  map<string, GossipInfo> infos = 1;
```

Comment:
Yeah, sounds good to me. Having looked at the gossip stuff more closely now, it seems obvious that we're going to need to use maps to achieve full-protoness.